CMakeLists.txt

```diff
@@ -1,73 +1,58 @@
-cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
-
-message(AUTHOR_WARNING "If the build fails, please try the autotools/qmake method.")
-
-if (POLICY CMP0074)
-    cmake_policy(SET CMP0074 NEW)
-endif()
-
-# TODO: fix the macOS bundle stuff
-if (POLICY CMP0080)
-    cmake_policy(SET CMP0080 OLD)
-endif()
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR) # Policies <= CMP0097 default to NEW
```

Comment:
I think this line is redundant given the recent conscious about requiring the latest version of cmake.
The specific `3.16` doesn't really mean anything specific for that cmake version, isn't? Then it is better to remove this line altogether instead of having a dummy number that will mislead one thinking there is something special with it.

Reply:
@Chocobo1 

> I think this line is redundant given the recent conscious about requiring the latest version of cmake.

I don't think that was the decision. We all settled with `3.16` as a compromise, for now. Besides, every CMake project should begin with `cmake_minimum_required`. Otherwise, users will be able to run the script with any version, but will get all sorts of errors and/or strange behavior if the version is not actually suitable.

When I proposed "always use the latest possible version", it really means "update `cmake_minimum_required` periodically, as it becomes convenient to support features/policy revisions introduced in newer versions, without regard for what versions are available in distro repositories".

Reply:
Is `3.16` really the required minimum version? a slightly older version won't work? If the answer is no for the first question, then I haven't seen you address my concern about it, quoting myself:
> ... mislead one thinking there is something special with it.

... otherwise I misunderstood and please disregard this post.

>We all settled with 3.16 as a compromise, for now. 
>as it becomes convenient to make use of features/policy revisions introduced in newer versions

Well I don't care how recent version we are using however I think it is a really bad compromise when the version doesn't really reflect the reality. I just want to point it out that maybe this can be overlooked for now but it *will* be a concern (at least from me) when we promote cmake build as officially supported.

Reply:
The ideal option from my point of view is that cmake_minimum_required really reflects the minimum supported version of CMake (but I do not strictly insist on this until CMake is the main build system of the project so I don't want to waste my time on this topic), otherwise the user may find itself in a situation where we require it to install a version really larger than the one that could successfully process our project (it may already have a suitable version installed, but it will fail due to this unreasonable restriction).

Reply:
>but I do not strictly insist on this until CMake is the main build system of the project

Agree.

>otherwise the user may find itself in a situation where we require it to install a version really larger than the one that could successfully process our project 

Agree.